src/output/elasticsearch/elasticsearch.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable sonarjs/no-identical-functions */
+import * as EL from "@elastic/elasticsearch";
+import { GetResponse } from "@elastic/elasticsearch/lib/api/types";
+import { defaultLog } from "@pagopa/winston-ts";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import * as B from "fp-ts/boolean";
+import { constVoid, pipe } from "fp-ts/lib/function";
+
+export interface IOutputDocument {
+  readonly [key: string]: unknown;
+  readonly id: string;
+}
+
+export interface IElasticError {
+  readonly statusCode: number;
+  readonly description: string;
+}
+
+export const toIndexError = (err: EL.errors.ResponseError): IElasticError => ({
+  description: err.message,
+  statusCode: err.statusCode
+});
```

Comment:
We can use a type guard instead

```suggestion
const isResponseError = (err: unknown): err is EL.errors.ResponseError =>
  err instanceof EL.errors.ResponseError;

export const toIndexError = (err: unknown): IElasticError =>
  pipe(
    err,
    E.fromPredicate(isResponseError, () => ({
      description: String(err),
      statusCode: 500
    })),
    E.map(responseError => ({
      description: responseError.message,
      statusCode: responseError.statusCode
    })),
    E.toUnion
  );
```